src/NuGet.Core/NuGet.Common/ZipArchiveExtensions.cs

```diff
@@ -88,13 +88,7 @@ public static void ExtractFiles(this ZipArchive archive, string targetPath, Func
                         Directory.CreateDirectory(targetEntryPath);
                     }
 
-                    using (var entryStream = entry.Open())
-                    {
-                        using (var targetStream = new FileStream(targetFile, FileMode.Create, FileAccess.Write, FileShare.None))
-                        {
-                            entryStream.CopyTo(targetStream);
-                        }
-                    }
+                    ZipFileExtensions.ExtractToFile(entry, targetFile, /*overwrite*/false);
```

Comment:
I believe overwrite should be true (for the case where packages have partially restored and we want to overwrite them)